backend/flags/flags.go

```diff
@@ -0,0 +1,119 @@
+package flags
+
+import (
+	"context"
+	"flag"
+	"reflect"
+	"time"
+
+	"github.com/heetch/confita"
+	"github.com/pkg/errors"
+)
+
+// Backend that loads configuration from the command line flags.
+type Backend struct {
+	parseFlagFn func()
+}
+
+// NewBackend creates a flags backend.
+func NewBackend() *Backend {
+	return &Backend{
+		parseFlagFn: flag.Parse,
+	}
+}
+
+// LoadStruct takes a struct config, define flags based on it and parse the command line args.
+func (b *Backend) LoadStruct(ctx context.Context, cfg *confita.StructConfig) error {
+	for _, field := range cfg.Fields {
+		f := field
+
+		if f.Backend != "" && f.Backend != b.Name() {
+			continue
+		}
+
+		// Display all the flags and their default values but override the field only if the user has explicitely
+		// set the flag.
+		k := f.Value.Kind()
+		switch {
+		case f.Value.Type().String() == "time.Duration":
+			// define the flag and its default value
+			v := flag.Duration(f.Key, time.Duration(f.Default.Int()), "")
+			// this function must be executed after the flag.Parse call.
+			defer func() {
+				// if the user has set the flag, save the value in the field.
+				if isFlagSet(f.Key) {
+					f.Value.SetInt(int64(*v))
+				}
+			}()
+		case k == reflect.Bool:
+			v := flag.Bool(f.Key, f.Default.Bool(), "")
+			defer func() {
+				if isFlagSet(f.Key) {
+					f.Value.SetBool(*v)
+				}
+			}()
+		case k >= reflect.Int && k <= reflect.Int64:
+			v := flag.Int(f.Key, int(f.Default.Int()), "")
+			defer func() {
+				if isFlagSet(f.Key) {
+					f.Value.SetInt(int64(*v))
+				}
+			}()
+		case k >= reflect.Uint && k <= reflect.Uint64:
+			v := flag.Uint(f.Key, uint(f.Default.Uint()), "")
+			defer func() {
+				f.Value.SetUint(uint64(*v))
+			}()
+		case k >= reflect.Float32 && k <= reflect.Float64:
+			v := flag.Float64(f.Key, f.Default.Float(), "")
+			defer func() {
+				f.Value.SetFloat(*v)
+			}()
+		case k == reflect.String:
+			v := flag.String(f.Key, f.Default.String(), "")
+			defer func() {
+				f.Value.SetString(*v)
+			}()
+		default:
+			flag.Var(&flagValue{f}, f.Key, "")
+		}
+	}
+
+	b.parseFlagFn()
```

Comment:
Why do you need to store the `flag.Parse` function into the field ? 
Why don't call directly `flag.Parse()`

Reply:
It was a left-over of a previous technique i used to test the flags. But since i switched to calling the test binary itself it's not needed anymore :+1: